src/main/java/frc/robot/commands/IntakeSequence.java

```diff
@@ -1,54 +1,96 @@
-        isTag = intakeLime.checkForTag();
-        SequentialCommandGroup moveArmElevatorClaw = new SequentialCommandGroup(
-            /*TODO do drive logic here*/
-            // the next two need to be done at the same time as the driving
+    public static Transform2d robotToTag;
+    public static Pose2d targetPose;
+  public IntakeSequence(
+      Claw claw, IntakeLimelight intakeLime, Arm arm, Elevator elevator, Climb climb, ScoringLimelight scoringLimelight, RobotContainer.Location location, DriveSubsystem drive) {
+    /* mechanical intake sequence - slightly modified version of what ruben had previously */
+    SequentialCommandGroup moveArmElevatorClaw =
+        new SequentialCommandGroup(
             new ConditionalCommand(
-                    new SequentialCommandGroup(
-                        new InstantCommand(() -> climb.setDesiredClimbPosition(ClimbPosition.CLEAR_OF_ARM)),
-                        new InstantCommand(() -> arm.setDesiredPosition(ArmPosition.INTAKE))
-                    ),
-                    new InstantCommand(() -> arm.setDesiredPosition(ArmPosition.INTAKE)), 
-                    () -> {return arm.isArmMoveable();}),
+                new SequentialCommandGroup(
+                    new InstantCommand(
+                        () -> climb.setDesiredClimbPosition(ClimbPosition.CLEAR_OF_ARM)),
+                    new WaitUntilCommand(()->climb.atDesiredPosition()),
+                    new InstantCommand(() -> arm.setDesiredPosition(ArmPosition.INTAKE))),
+                new InstantCommand(() -> arm.setDesiredPosition(ArmPosition.INTAKE)),
+                ()->arm.isArmInInterferenceZone()),
             new InstantCommand(() -> elevator.setDesiredPosition(ElevatorHeight.HOME)),
             new InstantCommand(() -> claw.runMotorsIntaking()),
             new WaitUntilCommand(claw::beamBreakSeesObject),
-            new InstantCommand(() -> claw.stopMotors())
-        );
-        addRequirements(claw);
-        addCommands(
-            new ConditionalCommand(
-                                    new SequentialCommandGroup(
-                                        /*TODO do drive logic here*/
-                                        // the next two need to be done at the same time as the driving
-                                        moveArmElevatorClaw
-                                    ), 
-                                    moveArmElevatorClaw, // return out of the command if the robot does not see the intake april tag 
-                                    () -> {return (!isTag.isEmpty() && ((tagid == 1) || (tagid == 2) || (tagid == 12) || (tagid == 13)));}
-                                )
-        );
-    }
+            new InstantCommand(() -> claw.stopMotors()));
+
+    BooleanSupplier checkForTag = () -> {
+      Optional<Transform2d> robotToTagOptional = scoringLimelight.checkForTag();
```

Comment:
am I stupid or should this use the intakeLimelight because this is the IntakeSequence ... ?